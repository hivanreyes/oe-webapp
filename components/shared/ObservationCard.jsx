import React from 'react'
import TruncateMarkup from 'react-truncate-markup'
import style from './styles/observationCard.scss'
import PropTypes from 'prop-types'

const ObservationCard = ({ data }) => {
  let postLabel = 'post'
  if (data.commentsCount !== 1) postLabel = 'posts'
  return (
    <div className={style.containerObsCard}>
      <div className={style.content}>
        <div className={style.titleCard}>
          <TruncateMarkup lines={1}>
            <div>{data.name}</div>
          </TruncateMarkup>
        </div>
        <div className={style.subTitle}>
          <TruncateMarkup lines={1}>
            <div>
              <span>{data.locationName}</span>
            </div>
          </TruncateMarkup>
        </div>
        <div className={style.description}>
          <TruncateMarkup lines={4}>
            <div>
              <span>{data.shareText.replace(/<(?:.|\n)*?>/gm, '')}</span>
            </div>
          </TruncateMarkup>
        </div>
        <img className={style.image} src={data.media} alt={data.name} />
        <div className={style.post}>
          <span className={style.number}>{data.commentsCount}</span>
          <span className={style.postText}>{postLabel}</span>
        </div>
      </div>
    </div>
  )
}

ObservationCard.propTypes = {
  data: PropTypes.shape({
    commentsCount: PropTypes.number.isRequired,
    locationName: PropTypes.string.isRequired,
    shareText: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    media: PropTypes.string.isRequired,
  }),
}

export default ObservationCard
