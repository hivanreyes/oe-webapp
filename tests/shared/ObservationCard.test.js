import React from 'react'
import { shallow } from 'enzyme'
import ObservationCard from '../../components/shared/ObservationCard'

const mockData = (commentsCount = 5) => ({
  data: {
    name: 'Expedition name',
    media: 'banner_url',
    shareText:
      '<b>Observation</b> <i>description</i>' +
      `<script>
         alert('alert')
       </script>`,
    commentsCount,
    locationName: 'Sausalito, California',
    duration: '04 April, 1997',
  },
})

describe('<ExpeditionCard />', () => {
  it('should shallow without errors', () => {
    const card = shallow(<ObservationCard {...mockData()} />)
    expect(card).toMatchSnapshot()
  })
  it('should strip html tags in its description', () => {
    const card = shallow(<ObservationCard {...mockData()} />)
    const description = card.find('.description span').props().children
    const strippedHTML = `Observation description
         alert('alert')
       `
    expect(description).toBe(strippedHTML)
  })
  it('should render `post` when postCount = 1', () => {
    const card = shallow(<ObservationCard {...mockData(1)} />)
    const countLabel = card.find('.postText').props().children
    expect(countLabel).toBe('post')
  })
  it('should render `posts` when postCount > 1', () => {
    const card = shallow(<ObservationCard {...mockData(2)} />)
    const countLabel = card.find('.postText').props().children
    expect(countLabel).toBe('posts')
  })
})
