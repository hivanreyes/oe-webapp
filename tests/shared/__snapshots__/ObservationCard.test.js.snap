// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpeditionCard /> should shallow without errors 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ObservationCard
    data={
      Object {
        "commentsCount": 5,
        "duration": "04 April, 1997",
        "locationName": "Sausalito, California",
        "media": "banner_url",
        "name": "Expedition name",
        "shareText": "<b>Observation</b> <i>description</i><script>
         alert('alert')
       </script>",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content"
      >
        <div
          className="titleCard"
        >
          <TruncateMarkup
            ellipsis="..."
            lineHeight=""
            lines={1}
          >
            <div>
              Expedition name
            </div>
          </TruncateMarkup>
        </div>
        <div
          className="subTitle"
        >
          <TruncateMarkup
            ellipsis="..."
            lineHeight=""
            lines={1}
          >
            <div>
              <span>
                Sausalito, California
              </span>
            </div>
          </TruncateMarkup>
        </div>
        <div
          className="description"
        >
          <TruncateMarkup
            ellipsis="..."
            lineHeight=""
            lines={4}
          >
            <div>
              <span>
                Observation description
         alert('alert')
       
              </span>
            </div>
          </TruncateMarkup>
        </div>
        <img
          alt="Expedition name"
          className="image"
          src="banner_url"
        />
        <div
          className="post"
        >
          <span
            className="number"
          >
            5
          </span>
          <span
            className="postText"
          >
            posts
          </span>
        </div>
      </div>,
      "className": "containerObsCard",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="titleCard"
          >
            <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={1}
            >
              <div>
                Expedition name
              </div>
            </TruncateMarkup>
          </div>,
          <div
            className="subTitle"
          >
            <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={1}
            >
              <div>
                <span>
                  Sausalito, California
                </span>
              </div>
            </TruncateMarkup>
          </div>,
          <div
            className="description"
          >
            <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={4}
            >
              <div>
                <span>
                  Observation description
         alert('alert')
       
                </span>
              </div>
            </TruncateMarkup>
          </div>,
          <img
            alt="Expedition name"
            className="image"
            src="banner_url"
          />,
          <div
            className="post"
          >
            <span
              className="number"
            >
              5
            </span>
            <span
              className="postText"
            >
              posts
            </span>
          </div>,
        ],
        "className": "content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={1}
            >
              <div>
                Expedition name
              </div>
            </TruncateMarkup>,
            "className": "titleCard",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                Expedition name
              </div>,
              "ellipsis": "...",
              "lineHeight": "",
              "lines": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Expedition name",
              },
              "ref": null,
              "rendered": "Expedition name",
              "type": "div",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={1}
            >
              <div>
                <span>
                  Sausalito, California
                </span>
              </div>
            </TruncateMarkup>,
            "className": "subTitle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                <span>
                  Sausalito, California
                </span>
              </div>,
              "ellipsis": "...",
              "lineHeight": "",
              "lines": 1,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Sausalito, California
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sausalito, California",
                },
                "ref": null,
                "rendered": "Sausalito, California",
                "type": "span",
              },
              "type": "div",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={4}
            >
              <div>
                <span>
                  Observation description
         alert('alert')
       
                </span>
              </div>
            </TruncateMarkup>,
            "className": "description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                <span>
                  Observation description
         alert('alert')
       
                </span>
              </div>,
              "ellipsis": "...",
              "lineHeight": "",
              "lines": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Observation description
         alert('alert')
       
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Observation description
         alert('alert')
       ",
                },
                "ref": null,
                "rendered": "Observation description
         alert('alert')
       ",
                "type": "span",
              },
              "type": "div",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Expedition name",
            "className": "image",
            "src": "banner_url",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="number"
              >
                5
              </span>,
              <span
                className="postText"
              >
                posts
              </span>,
            ],
            "className": "post",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 5,
                "className": "number",
              },
              "ref": null,
              "rendered": 5,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "posts",
                "className": "postText",
              },
              "ref": null,
              "rendered": "posts",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content"
        >
          <div
            className="titleCard"
          >
            <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={1}
            >
              <div>
                Expedition name
              </div>
            </TruncateMarkup>
          </div>
          <div
            className="subTitle"
          >
            <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={1}
            >
              <div>
                <span>
                  Sausalito, California
                </span>
              </div>
            </TruncateMarkup>
          </div>
          <div
            className="description"
          >
            <TruncateMarkup
              ellipsis="..."
              lineHeight=""
              lines={4}
            >
              <div>
                <span>
                  Observation description
         alert('alert')
       
                </span>
              </div>
            </TruncateMarkup>
          </div>
          <img
            alt="Expedition name"
            className="image"
            src="banner_url"
          />
          <div
            className="post"
          >
            <span
              className="number"
            >
              5
            </span>
            <span
              className="postText"
            >
              posts
            </span>
          </div>
        </div>,
        "className": "containerObsCard",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="titleCard"
            >
              <TruncateMarkup
                ellipsis="..."
                lineHeight=""
                lines={1}
              >
                <div>
                  Expedition name
                </div>
              </TruncateMarkup>
            </div>,
            <div
              className="subTitle"
            >
              <TruncateMarkup
                ellipsis="..."
                lineHeight=""
                lines={1}
              >
                <div>
                  <span>
                    Sausalito, California
                  </span>
                </div>
              </TruncateMarkup>
            </div>,
            <div
              className="description"
            >
              <TruncateMarkup
                ellipsis="..."
                lineHeight=""
                lines={4}
              >
                <div>
                  <span>
                    Observation description
         alert('alert')
       
                  </span>
                </div>
              </TruncateMarkup>
            </div>,
            <img
              alt="Expedition name"
              className="image"
              src="banner_url"
            />,
            <div
              className="post"
            >
              <span
                className="number"
              >
                5
              </span>
              <span
                className="postText"
              >
                posts
              </span>
            </div>,
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TruncateMarkup
                ellipsis="..."
                lineHeight=""
                lines={1}
              >
                <div>
                  Expedition name
                </div>
              </TruncateMarkup>,
              "className": "titleCard",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div>
                  Expedition name
                </div>,
                "ellipsis": "...",
                "lineHeight": "",
                "lines": 1,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Expedition name",
                },
                "ref": null,
                "rendered": "Expedition name",
                "type": "div",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TruncateMarkup
                ellipsis="..."
                lineHeight=""
                lines={1}
              >
                <div>
                  <span>
                    Sausalito, California
                  </span>
                </div>
              </TruncateMarkup>,
              "className": "subTitle",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div>
                  <span>
                    Sausalito, California
                  </span>
                </div>,
                "ellipsis": "...",
                "lineHeight": "",
                "lines": 1,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Sausalito, California
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sausalito, California",
                  },
                  "ref": null,
                  "rendered": "Sausalito, California",
                  "type": "span",
                },
                "type": "div",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TruncateMarkup
                ellipsis="..."
                lineHeight=""
                lines={4}
              >
                <div>
                  <span>
                    Observation description
         alert('alert')
       
                  </span>
                </div>
              </TruncateMarkup>,
              "className": "description",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div>
                  <span>
                    Observation description
         alert('alert')
       
                  </span>
                </div>,
                "ellipsis": "...",
                "lineHeight": "",
                "lines": 4,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Observation description
         alert('alert')
       
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Observation description
         alert('alert')
       ",
                  },
                  "ref": null,
                  "rendered": "Observation description
         alert('alert')
       ",
                  "type": "span",
                },
                "type": "div",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Expedition name",
              "className": "image",
              "src": "banner_url",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="number"
                >
                  5
                </span>,
                <span
                  className="postText"
                >
                  posts
                </span>,
              ],
              "className": "post",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 5,
                  "className": "number",
                },
                "ref": null,
                "rendered": 5,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "posts",
                  "className": "postText",
                },
                "ref": null,
                "rendered": "posts",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
